src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        ColorSupplier colorSupplier = new ColorSupplier();

```

Comment:
Each time this method is called, a new ColorSupplier instance will be created. Let's think about how we can optimize this logic and create an instance only once.